Model/ConnectionTypeResolver.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ *  @package BelVG AWS Sqs.
+ *  @copyright 2018
+ *
+ */
+namespace Belvg\Sqs\Model;
+
+use Magento\Framework\MessageQueue\ConnectionTypeResolverInterface;
+use Magento\Framework\App\DeploymentConfig;
+
+/**
+ * AWS SQS connection type resolver.
+ *
+ * @api
+ */
+class ConnectionTypeResolver implements ConnectionTypeResolverInterface
+{
+    /**
+     * AWS SQS connection names.
+     *
+     * @var string[]
+     */
+    private $sqsConnectionName = [];
+
+    /**
+     * Initialize dependencies.
+     *
+     * @param DeploymentConfig $deploymentConfig
+     * @since 100.0.0
+     */
+    public function __construct(DeploymentConfig $deploymentConfig)
+    {
+        $queueConfig = $deploymentConfig->getConfigData(Config::QUEUE_CONFIG);
+        if (isset($queueConfig['connections']) && is_array($queueConfig['connections'])) {
+            $this->sqsConnectionName = array_keys($queueConfig['connections']);
+        }
+        if (isset($queueConfig[Config::SQS_CONFIG])) {
+            $this->sqsConnectionName[] = Config::SQS_CONFIG;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     * @since 100.0.0
+     */
+    public function getConnectionType($connectionName)
+    {
+        return in_array($connectionName, $this->sqsConnectionName) ? 'sqs' : null;
```

Comment:
Use, please, so:
```php
return isset($this->sqsConnectionName[$connectionName]) ? 'sqs' : null;
```
It will be executed much faster, without overhead

Reply:
Changed

Reply:
I reviewed this again, but this: 

return isset(($this$this->->sqsConnectionNamesqsConnectionNam [$connectionName]) ? 'sqs' : null;

Is different then: 

return in_array($connectionName, $this->sqsConnectionName) ? 'sqs' : null;

The array sqsConnectionName holds the names of the connections as values of that array. So the in_array function which is used here is correct, which checks for the values. isset will check the array key and returns null in this case.